static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +537,319 @@ export function getProteinDescription(data: GraphNodes): string {
+  });
+}
+
+/**
+ * Convert DCID of form bio/<id> to id.
+ * Note: probably should be doing dcid: `bio/${string}`, but currently causes many type errors
+ */
+export function idFromDCID(dcid: string): string {
+  return dcid.replace("bio/", "");
+}
+
+/**
+ * Given id, convert to DCID of form bio/<id>
+ */
+export function dcidFromID(id: string): string {
+  return `bio/${id}`;
+}
+
+/**
+ * Given quantity DCID of the form "<quantityName><quantityValue>", extract and return quantityValue.
+ */
+export function quantityFromDCID(
+  quantityDCID: string,
+  quantityName: string
+): number {
+  if (!quantityDCID.includes(quantityName)) {
+    throw `DCID "${quantityDCID}" does not contain quantity name "${quantityName}"`;
+  }
+  return Number(_.last(quantityDCID.split(quantityName)));
+}
+
+/**
+ * Given interaction DCID of the form bio/{protein1}_{protein2}, return [protein1, protein2]
+ */
+export function proteinsFromInteractionDCID(interactionDCID: string): string[] {
+  const id = idFromDCID(interactionDCID);
+  // danger: assumes neither {protein id}, {species id} contain an underscore
+  const split = id.split("_");
+  if (split.length !== 4) {
+    throw `Unsupported DCID ${interactionDCID}`;
+  }
+  return [`${split[0]}_${split[1]}`, `${split[2]}_${split[3]}`];
+}
+
+/**
+ * Given interactionDCID of the form {protein 1}_{protein 2}, return ID of the same form
+ * but with protein 1, protein 2 in alpha order.
+ */
+function sortInteractionDCID(interactionDCID: string): string {
+  const proteins = proteinsFromInteractionDCID(interactionDCID);
+  // sort proteins alphabetically so A_B and B_A map to the same ID A_B
+  const proteinsSorted = proteins.sort((p1, p2) => p1.localeCompare(p2));
+  return dcidFromID(proteinsSorted.join("_"));
+}
+
+/**
+ * Given list of interaction DCIDs, for each subset of DCIDs identifying the same pair of proteins (e.g. A_B and B_A),
+ * keep only one.
+ */
+export function deduplicateInteractionDCIDs(
+  interactionDCIDs: string[]
+): string[] {
+  // Can't just use Set here because not guaranteed that A_B in list implies B_A in list,
+  // so we need to keep track of which order is the true DCID corresponding to an entity in the response.
+  // Counterexample: http://datacommons.org/browser/bio/KLOTB_HUMAN
+  return Object.values(
+    objectFromArray(interactionDCIDs, sortInteractionDCID, (dcid) => dcid)
+  ) as string[];
+}
+
+/**
+ * Given interaction DCID and source DCID, infer and return target ID (or target DCID if returnDCID is true).
+ */
+export function getInteractionTarget(
+  interactionDCID: string,
+  sourceDCID: string,
+  returnDCID = false
+): string {
+  // note this also works in the case of a self-interaction
+  const interactionID = idFromDCID(interactionDCID);
+  const sourceID = idFromDCID(sourceDCID);
+  const id = interactionID
+    .replace(`${sourceID}_`, "")
+    .replace(`_${sourceID}`, "");
+  if (returnDCID) return dcidFromID(id);
```

Comment:
For ifs with 1-line bodies, do you guys prefer 1-line or multiple lines?

Reply:
multiple lines please, for readability.

Reply:
got it, thanks!